Snowflake.Data/Core/Tools/EnvironmentOperations.cs

```diff
@@ -13,9 +13,9 @@ internal class EnvironmentOperations
         public static readonly EnvironmentOperations Instance = new EnvironmentOperations();
         private static readonly SFLogger s_logger = SFLoggerFactory.GetLogger<EnvironmentOperations>();
 
-        public virtual string GetEnvironmentVariable(string variable)
+        public virtual string GetEnvironmentVariable(string variable, string defaultValue = null)
```

Comment:
The idea of overloading this method is to avoid checking in each scenario if the environment variable is not defined when we want to have a default value. This simplifies our code when it requires a default value. Additionally, for mocking, you only need to specify the second parameter as null to work as now. The default value is an optional parameter, so it will not generate any issues.

https://github.com/snowflakedb/snowflake-connector-net/blob/99a1d4c67487f2e9d1abc2487c3f86a3bf1a029f/Snowflake.Data.Tests/UnitTests/Configuration/EasyLoggingConfigFinderTest.cs#L241

Reply:
Changed to handle each default value in GetEnvironment specific use.